src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,127 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int MINIMUM_LENGTH_ARRAY = 10;
+
+    private static final double CAPACITY_MULTIPLIER = 1.5;
+
+    private T[] values = (T[]) new Object[MINIMUM_LENGTH_ARRAY];
+    private int size = 0;
+
     @Override
     public void add(T value) {
-
+        if (size() == values.length) {
+            grow();
+        }
+        values[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index == size()) {
+            add(value);
+            return;
+        }
+        checkNullAndSize(index);
+        if (size() == values.length) {
```

Comment:
use size variable directly instead of calling method

Reply:
not fixed